src/client/conn/http2.rs

```diff
@@ -60,13 +68,15 @@ pub struct Builder {
 pub async fn handshake<E, T, B>(
     exec: E,
     io: T,
-) -> crate::Result<(SendRequest<B>, Connection<T, B>)>
+) -> crate::Result<(SendRequest<B>, Connection<T, B, E>)>
 where
     E: Executor<BoxSendFuture> + Send + Sync + 'static,
     T: AsyncRead + AsyncWrite + Unpin + Send + 'static,
     B: Body + 'static,
     B::Data: Send,
     B::Error: Into<Box<dyn StdError + Send + Sync>>,
+    E: ExecutorClient<B, T> + Unpin + Clone,
+    <B as http_body::Body>::Error: std::error::Error + Send + Sync + 'static,
```

Comment:
The line just above, `B::Error: Into<..>` should be enough. Is it not?

Reply:
Is it better now?

Reply:
Is it required to add something here at all? Because currently, this makes it stricter in that not only is it `Into`, but it must also _be_ `Error`. That means `Box<dyn Error>` no longer qualifies (since `Box<dyn Error>: Error` is not true for confusing reasons).